test/src/main/java/org/apache/accumulo/test/functional/IteratorMincClassCastBugIT.java

```diff
@@ -0,0 +1,516 @@
+      ntc.setLocalityGroups(groups);
+
+      IteratorSetting iteratorSetting = new IteratorSetting(20, ChunkCombiner.class);
+      ntc.attachIterator(iteratorSetting, EnumSet.of(IteratorUtil.IteratorScope.minc));
+
+      c.tableOperations().create(tableName, ntc);
+
+      int chunkSize = 64 * 1024;
+      ColumnVisibility visibility = new ColumnVisibility();
+
+      List<URL> files = new ArrayList<>();
+      files.add(getClass().getClassLoader().getResource("testfile1.md"));
+
+      try (BatchWriter bw = c.createBatchWriter(tableName, new BatchWriterConfig())) {
+        FileDataIngest fdi = new FileDataIngest(chunkSize, visibility);
+        for (URL filename : files) {
+          fdi.insertFileData(filename, bw);
+        }
+      }
+
+      c.tableOperations().flush(tableName, null, null, true);
+    }
+  }
+
+  /**
+   * Copied from 2.0 examples. Takes a list of files and archives them into Accumulo keyed on hashes
+   * of the files.
+   */
+  @SuppressFBWarnings(value = "WEAK_MESSAGE_DIGEST_MD5", justification = "For testing only")
+  public static class FileDataIngest {
+    public static final Text CHUNK_CF = new Text("~chunk");
+    public static final Text REFS_CF = new Text("refs");
+    public static final String REFS_ORIG_FILE = "name";
+    public static final String REFS_FILE_EXT = "filext";
+    public static final ByteSequence CHUNK_CF_BS =
+        new ArrayByteSequence(CHUNK_CF.getBytes(), 0, CHUNK_CF.getLength());
+    public static final ByteSequence REFS_CF_BS =
+        new ArrayByteSequence(REFS_CF.getBytes(), 0, REFS_CF.getLength());
+
+    int chunkSize;
+    byte[] chunkSizeBytes;
+    byte[] buf;
+    MessageDigest md5digest;
+    ColumnVisibility cv;
+
+    public FileDataIngest(int chunkSize, ColumnVisibility colvis) {
+      this.chunkSize = chunkSize;
+      chunkSizeBytes = intToBytes(chunkSize);
+      buf = new byte[chunkSize];
+      try {
+        md5digest = MessageDigest.getInstance("MD5");
+      } catch (NoSuchAlgorithmException e) {
+        throw new RuntimeException(e);
+      }
+      cv = colvis;
+    }
+
+    public String insertFileData(URL fileURL, BatchWriter bw)
+        throws MutationsRejectedException, IOException {
+      String filename = fileURL.getFile();
+      if (chunkSize == 0)
+        return "";
+      md5digest.reset();
+      String uid = hexString(md5digest.digest(filename.getBytes()));
+
+      // read through file once, calculating hashes
+      md5digest.reset();
+      InputStream fis = null;
+      int numRead = 0;
+      // try (var reader = new InputStreamReader(accumuloPropsLocation.openStream(), UTF_8)) {
+      try {
+        fis = new FileInputStream(filename);
+        numRead = fis.read(buf);
+        while (numRead >= 0) {
+          if (numRead > 0) {
+            md5digest.update(buf, 0, numRead);
+          }
+          numRead = fis.read(buf);
+        }
+      } finally {
+        if (fis != null) {
+          fis.close();
+        }
+      }
```

Comment:
Our test file is small enough that this can be reduced to two simple lines... one to read the file fully into a String, and another to compute the digest from the String's bytes. This would improve the comprehensibility of the test, since we don't actually need to chunk and buffer the tiny test file.

Reply:
I rewrote the IT so it doesn't need to ingest a file now.